src/SystemReport.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace Krokedil\Support;
+
+use Automattic\WooCommerce\Internal\Admin\Logging\Settings;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class SystemReport
+ *
+ * @package Krokedil\Support
+ */
+class SystemReport {
+
+	public const REGISTRY_OPTION = 'krokedil_support_registry';
+
+	/**
+	 * The gateway ID.
+	 *
+	 * @var string
+	 */
+	private $id;
+
+	/**
+	 * The gateway name (or title).
+	 *
+	 * @var string
+	 */
+	private $name;
+
+	private $included = array();
+	private $excluded = array();
+
+	/**
+	 * SystemReport constructor.
+	 *
+	 * @param string $id The gateway ID.
```

Comment:
This is a overall thing through most of the code, you have defined these as "gateway id" or "gateway name". Id recommend changing this to cover more types of plugins, so a generic "plugin id" and "plugin name" instead might be a good idea.